crates/rpc/rpc-engine-api/src/engine_api.rs

```diff
@@ -265,25 +267,14 @@ where
         )?;
 
         // Now resolve the payload
-        let mut resolved_payload = self
-            .inner
+        self.inner
             .payload_store
             .resolve(payload_id)
             .await
             .ok_or(EngineApiError::UnknownPayload)?
-            .map(|payload| payload.into_v3_payload())?;
-
-        // After `Cancun` is enabled on optimism, an extra field `parent_beacon_block_root` is
-        // included in the enveloped V3 payload. On ethereum, this field is not included.
-        if self.inner.chain_spec.is_optimism() &&
-            self.inner
-                .chain_spec
-                .is_fork_active_at_timestamp(Hardfork::Cancun, attributes.timestamp())
-        {
-            resolved_payload.parent_beacon_block_root = attributes.parent_beacon_block_root();
-        }
```

Comment:
ah I see now that we're using TryInto, we can no longer do this manually,

though we need this for Optimism, see my other comment about `OptimismBuiltPayload`, 
I think we need to add this type in this PR as well.

we could however solve this in two ways I believe, either add a new wrapper type `OptimismBuiltPayload` or we also include the `PayloadAttributes` in:
https://github.com/paradigmxyz/reth/blob/main/crates/payload/builder/src/payload.rs#L26-L26 and add the conversion for `OptimismExecutionPayloadV3` that has the parent_beacon_block_root

perhaps we actually want both, 
in any case I think we need the `OptimismExecutionPayloadV3` type

Reply:
> in any case I think we need the `OptimismExecutionPayloadV3` type

ok, proposed here https://github.com/alloy-rs/alloy/pull/242 ptal